snuba/query/processors/join_optimizers.py

```diff
@@ -0,0 +1,44 @@
+from snuba.datasets.schemas.join import JoinClause
+from snuba.query.columns import all_referenced_columns, QUALIFIED_COLUMN_REGEX
+from snuba.query.query import Query
+from snuba.query.query_processor import QueryProcessor
+from snuba.request.request_settings import RequestSettings
+
+
+class SimpleJoinOptimizer(QueryProcessor):
+    """
+    Simplest possible join optimizer. It turns a join expression into a single
+    table expression if only one table is referenced in the query.
+    At this stage this is basically a proof of concept, we can build
+    a more sophisticated optimizaiton based on this.
+
+    TODO: Optimize a join between multiple tables by minimizing the number
+    of tables joined together when more than one is referenced in the query.
+    """
+
+    def process_query(self,
+        query: Query,
+        request_settings: RequestSettings,
+    ) -> None:
+        from_clause = query.get_data_source()
+        if not isinstance(from_clause, JoinClause):
+            return
+
+        referenced_columns = all_referenced_columns(query)
+        referenced_aliases = set()
+        for qualified_column in referenced_columns:
+            # This will be much better when we will represent columns
+            # with a more structured data type than strings.
+            match = QUALIFIED_COLUMN_REGEX.match(qualified_column)
+            if match:
+                table_alias = match[1]
```

Comment:
Isn't this result `(table alias, column alias)`?

Reply:
no, it is 
(original expression, table alias, column alias):
```
>>> import re
>>> QUALIFIED_COLUMN_REGEX = re.compile(r"^([a-zA-Z_][a-zA-Z0-9_]*)\.([a-zA-Z0-9_\.\[\]]+)$")
>>> a = QUALIFIED_COLUMN_REGEX.match("a.b")
>>> a.match()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 're.Match' object has no attribute 'match'
>>> a[0]
'a.b'
>>> a[1]
'a'
>>> a[2]
'b'
>>> 
```